test/browser/webgl_offscreen_framebuffer_swap_with_bad_state.c

```diff
@@ -27,6 +27,13 @@ int main() {
 
   EMSCRIPTEN_WEBGL_CONTEXT_HANDLE ctx = emscripten_webgl_create_context("#canvas", &attr);
   emscripten_webgl_make_context_current(ctx);
+#ifdef TEST_DISABLE_VAO
+  EM_ASM({
+    console.log(GL.currentContext.defaultVao);
+    assert(GL.currentContext.defaultVao);
```

Comment:
This assert seems like it should fail on WebGL 1.0 because we won't have created a `defaultVao` unless OES_vertex_array_object has been enabled, which it hasn't yet (that happens below).

Except the code that _creates_ `defaultVao` runs early, in `createContext`>`registerContext`>`createOffscreenFramebuffer`, so evidently `webgl_enable_OES_vertex_array_object` has already run by that point.
I don't know how that happens.

Reply:
IIUC you figured this part out right?